code/game/machinery/chemHandC.dm

```diff
@@ -0,0 +1,264 @@
+
+//Heater
+
+/obj/machinery/chemheater
+	name = "directed laser heater"
+	desc = "A small stage where a container can be placed to heat it's contents with a laser."
+	icon = 'icons/obj/chemHandC.dmi'
+	icon_state = "heater"
+	icon_state_open = "heater_open"
+	density = 1
+	anchored = 1
+	machine_flags = SCREWTOGGLE | CROWDESTROY | WRENCHMOVE | FIXED2WORK
+	use_power = 1
+	idle_power_usage = 25
+	active_power_usage = 5000
+
+	var/max_temperature = TEMPERATURE_LASER
+	var/thermal_energy_transfer = 3000
+	var/laser_kind = 0
+	var/onstage = null
+
+	var/obj/item/weapon/reagent_containers/held_container
+	var/heating = 0
+	var/had_item = 0
+
+/obj/machinery/chemheater/New()
+	. = ..()
+
+	component_parts = newlist(
+		/obj/item/weapon/circuitboard/chemheater,
+		/obj/item/weapon/stock_parts/micro_laser,
+		/obj/item/weapon/stock_parts/capacitor
+	)
+	RefreshParts()
+	overlays += image(icon = icon, icon_state = "t[laser_kind]_laser") //Adds the side laser
+	update_icon()
+
+/obj/machinery/chemheater/RefreshParts()
+	var/T = 0
+	for(var/obj/item/weapon/stock_parts/micro_laser/L in component_parts)
+		T += L.rating
+		laser_kind = L.rating //Sets what tier of laser we have
+	thermal_energy_transfer = initial(thermal_energy_transfer) * T
+
+	T = 0
+	for(var/obj/item/weapon/stock_parts/capacitor/C in component_parts)
+		T += C.rating-1
+	idle_power_usage = initial(idle_power_usage) - (T * 10) //T1: 25w, T2: 15w, T3: 5w
+	active_power_usage = initial(active_power_usage) - (T * 2000) //T1: 5000w, T2: 3000w, T3: 1000w
+
+	overlays = null
+	overlays += image(icon = icon, icon_state = "t[laser_kind]_laser")
+	update_icon()
+
+/obj/machinery/chemheater/power_change()
+	if( powered() )
+		icon_state = initial(icon_state)
+		stat &= ~NOPOWER
+	else
+		spawn(rand(0, 15))
+			src.icon_state = "[initial(icon_state)]_off"
+			stat |= NOPOWER
+
+/obj/machinery/chemheater/process()
+	if(held_container && heating)
+		held_container.reagents.heating(thermal_energy_transfer, max_temperature)
+
+/obj/machinery/chemheater/attackby(obj/item/weapon/W, mob/user)
+	if(istype(W, /obj/item/weapon/reagent_containers) && anchored)
+		if(!held_container)
+			if(user.drop_item(W, src))
+				held_container = W
+				to_chat(user, "<span class='notice'>You put \the [held_container] onto \the [src].</span>")
+				var/image/I = image("icon"=W, "layer"=FLOAT_LAYER)
+				onstage = I
+				overlays += I
+				return 1
+		else
+			to_chat(user, "<span class='notice'>\The [src] already has \a [held_container] on it.</span>")
+			return 1
+	else
+		..()
```

Comment:
This should probably be `return ..()`, to notify the caller in case the parent returned 1